### ISSUE TITLE
testinterface/TestMainBase#testMain should check serverPort arg

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#1729]
OK, this comes under the rubric of "People who do not know what they are doing should not
mess with things".  I am reporting it because I believe I am at least the second
person to have stumbled onto this problem and I have probably stumbled on it more than once.

### Problem (What I am seeing):
Running the the target/scala-2.11/foo-out native image generated by " sbt> test"
fails with a correct but less than helpful exception:
```
$ ./foo-out
java.util.NoSuchElementException: next on empty iterator
	at java.lang.Throwable.fillInStackTrace(Unknown Source)
	at scala.collection.Iterator$$anon$2.next(Unknown Source)
	at scala.collection.Iterator$$anon$2.next(Unknown Source)
	at scala.collection.IndexedSeqLike$Elements.next(Unknown Source)
	at scala.collection.IterableLike$class.head(Unknown Source)
	at scala.collection.mutable.ArrayOps$ofRef.scala$collection$IndexedSeqOptimized$$super$head(Unknown Source)
	at scala.collection.IndexedSeqOptimized$class.head(Unknown Source)
	at scala.collection.mutable.ArrayOps$ofRef.head(Unknown Source)
	at scala.scalanative.testinterface.TestMainBase.testMain(Unknown Source)
	at scala.scalanative.testinterface.TestMain$.main(Unknown Source)
	at <none>.main(Unknown Source)
	at <none>.__libc_start_main(Unknown Source)
	at <none>._start(Unknown Source)
```
### Analysis
It is entirely reasonable, to me at least, to try running the foo-out test image as native.  Turns out
that testMain() has a requirement for an argument `serverPort` so running the native image will
fail. Fair enough, especially for a solo developer bootstrap implementation.
```
/** Actual main method of the test runner. */
  def testMain(args: Array[String]): Unit = {
    val serverPort   = args.head.toInt
    val clientSocket = new Socket("127.0.0.1", serverPort)
    testRunner(Array.empty, null, clientSocket)
  }
```

### Solution (what I would like to see):

I would like to see testMain() check that args.length == 1 and give a `usage: testMain serverPort`
or "testing must be done within sbt"
text message if not.  That is, gently guide a new and/or uninformed to the proper way to do things.
This avoids antagonizing potential & existing friends of ScalaNative.  If the trivial fix prevents
alienating even one devo, it is worth the effort/overhead.

### Next steps
If discussion here converges to some consensus text for the error message, I will file a PR. 
I welcome gentle comments and suggestions.

**Comment by sjrd** [comment]
A PR with a better error message, whatever it is, would be accepted. Although this is not part of the public, documented behavior, and we will get rid of the whole `TestMain` thing once #1728 is done. So maybe it's not worth the effort at this point.

**Comment by LeeTibbert** [comment]
@sjrd Thank you for your feedback.  Based on it, I will hang loose until #1728 is done and has had a chance to cure.   All things in their proper time.

**Comment by LeeTibbert** [comment]
@errikos kindly fixed this in PR #1802, since merged.  Thank you.
